src/types/rows.rs

```diff
@@ -18,374 +18,68 @@ pub struct Row {
-                    (Inet) is only supported.",
-                                               cassandra_type));
-                    Err(io_err)
-                }
-            };
-
-            return converted.map_err(|err| err.into());
-        });
-    }
-}
-
-impl IntoRustByName<Uuid> for Row {
-    fn get_by_name(&self, name: &str) -> Option<Result<Uuid>> {
-        return self.get_col_by_name(name).map(|(cassandra_type, cbytes)| {
-            let bytes = cbytes.as_slice();
-
-            if bytes.is_empty() {
-                return Err(column_is_empty_err());
-            }
-
-            let converted = match cassandra_type {
-                &ColType::Uuid => decode_timeuuid(bytes),
-                &ColType::Timeuuid => decode_timeuuid(bytes),
-                _ => unreachable!(),
-            };
-
-            return converted.map_err(|err| err.into());
-        });
-    }
-}
-
-impl IntoRustByName<List> for Row {
-    fn get_by_name(&self, name: &str) -> Option<Result<List>> {
-        return self.get_col_spec_by_name(name).map(|(cassandra_type, cbytes)| {
-            let bytes = cbytes.as_slice();
-
-            if bytes.is_empty() {
-                return Err(column_is_empty_err());
-            }
-
-            return match cassandra_type.col_type.id {
-                // in fact, both decode_list and decode_set return Ok
-                ColType::List => {
-                    Ok(List::new(decode_list(bytes).unwrap(), cassandra_type.col_type.clone()))
-                }
-                ColType::Set => {
-                    Ok(List::new(decode_set(bytes).unwrap(), cassandra_type.col_type.clone()))
-                }
-                _ => unreachable!(),
-            };
-        });
-    }
-}
-
-impl IntoRustByName<Map> for Row {
-    fn get_by_name(&self, name: &str) -> Option<Result<Map>> {
-        return self.get_col_spec_by_name(name).map(|(cassandra_type, cbytes)| {
-            let bytes = cbytes.as_slice();
-
-            if bytes.is_empty() {
-                return Err(column_is_empty_err());
-            }
-
-            return match cassandra_type.col_type.id {
-                // in fact, both decode_map and decode_set return Ok
-                ColType::Map => {
-                    Ok(Map::new(decode_map(bytes).unwrap(), cassandra_type.col_type.clone()))
-                }
-                _ => unreachable!(),
-            };
-        });
+        self.get_col_spec_by_name(name)
+            .map(|(col_spec, cbytes)| {
+                let ref col_type = col_spec.col_type;
+                as_rust!(col_type, cbytes, Vec<u8>)
+            })
     }
 }
 
-impl IntoRustByName<UDT> for Row {
-    fn get_by_name(&self, name: &str) -> Option<Result<UDT>> {
-        return self.get_col_spec_by_name(name).map(|(cassandra_type, cbytes)| {
-            let bytes = cbytes.as_slice();
+macro_rules! row_into_rust_by_name {
+    ($($into_type:tt)*) => (
```

Comment:
Maybe in one of future PRs we could have a single macro `into_rust_by_name` with few 'arms' 

```
macro_rules! into_rust_by_name {
    ($($into_type:tt)*, Row) => {},
    ($($into_type:tt)*, Map) => {},
    ($($into_type:tt)*, List) => {},
    ...
}
```

Then in a specific modules we could call it to deliver specific implemenations to specific data structures.